app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/pocket/PocketStoriesComposables.kt

```diff
@@ -2,188 +2,114 @@
-
-                Placeholder {
-                    Box(
-                        Modifier.background(
-                            when (isSystemInDarkTheme()) {
-                                true -> Color(0xFF42414D) // DarkGrey30
-                                false -> PhotonColors.LightGrey30
-                            }
-                        )
-                    )
-                }
-
-                Fallback {
-                    Box(
-                        Modifier.background(
-                            when (isSystemInDarkTheme()) {
-                                true -> Color(0xFF42414D) // DarkGrey30
-                                false -> PhotonColors.LightGrey30
-                            }
-                        )
-                    )
-                }
-            }
-        }
-
-        Spacer(modifier = Modifier.height(8.dp))
-
-        CompositionLocalProvider(LocalContentAlpha provides ContentAlpha.medium) {
-            Text(
-                modifier = Modifier.padding(bottom = 2.dp),
-                text = story.publisher,
-                style = MaterialTheme.typography.caption,
-                maxLines = 1,
-                overflow = TextOverflow.Ellipsis
-            )
+    val imageUrl = story.imageUrl.replace(
+        "{wh}",
+        with(LocalDensity.current) { "${116.dp.toPx().roundToInt()}x${84.dp.toPx().roundToInt()}" }
+    )
+    ListItemTabLarge(
+        client = client,
+        imageUrl = imageUrl,
+        onClick = { onStoryClick(story) },
+        title = { TabTitle(text = story.title, maxLines = 3) },
+        subtitle = {
+            TabSubtitleWithInterdot(story.publisher, "${story.timeToRead} min")
         }
-        Text(
-            text = story.title,
-            style = MaterialTheme.typography.subtitle1,
-            maxLines = 4,
-            overflow = TextOverflow.Ellipsis
-        )
-    }
+    )
 }
 
 /**
- * Displays a list of [PocketRecommendedStory]es.
+ * Displays a list of [PocketRecommendedStory]es on 3 by 3 grid.
+ * If there aren't enough stories to fill all columns placeholders containing an external link
+ * to go to Pocket for more recommendations are added.
+ *
+ * @param stories the list of [PocketRecommendedStory]ies to be displayed. Expect a list with 8 items.
+ * @param client [Client] instance to be used for downloading the story header image.
+ * @param onExternalLinkClicked callback for when the user taps an element which contains an
+ * external link for where user can go for more recommendations.
  */
 @Composable
 fun PocketStories(
     @PreviewParameter(PocketStoryProvider::class) stories: List<PocketRecommendedStory>,
-    client: Client
+    client: Client,
+    onExternalLinkClicked: (String) -> Unit
 ) {
-    // Items will be shown on two rows. Ceil the divide result to show more items on the top row.
-    val halfStoriesIndex = (stories.size + 1) / 2
+    // Show stories in a 3 by 3 grid
+    val gridLength = 3
```

Comment:
I don't know if this will useful, but have you seen https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/windowed.html

Reply:
That seems nice, thank you!
There is an update needed, some specs slightly changed regarding what stories to display. I'll try using this in that PR.